repos/fdbt-site/cypress_tests/cypress/support/steps.ts

```diff
@@ -117,6 +118,8 @@ export const completeFlatFareCarnet = (): void => {
 export const completeServicePage = (): void => {
     selectRandomOptionFromDropDown('service');
     continueButtonClick();
+    checkAndRepickService();
+    checkAndRepickService();
```

Comment:
So it will retry twice, what happens if the first three options all fail?

We probably want to keep iterating over all of the possible options until we run out

Reply:
We have an opportunity here to change the code to fail 0% of the time, assuming that one of the services setup for the `noc` will allow a return product, so I think that would be more robust that this. 

Unless we just want to ensure this services exists in the test seed data and we can do away with this retry and just choose the one we create from the dropdown.

If we want to continue on the retry approach, relying on Cypress/BrowserStack to retry will not only increase the overall test run time, but unless you can find out if the randomness in a test run is non-deterministic then the same random option might be chosen from the dropdown each time. Therefore we should exhaust all dropdown options in a single test run.

I think a bit of a refactor is needed to make this code flow better and be as robust as possible. We should look to remove the duplication of the lines:
```
    selectRandomOptionFromDropDown('service');
    continueButtonClick();
```
So rather than trying the `continueButtonClick()', then going to the retry functions in the error case, we do something like this pseudo code:
```
completeServicePage(){
  serviceOption = numberOfOptionsInServiceDropdown
  for serviceOption; do
    if "you cannot create a return product for this service" is not on the page;
      selectOptionFromDropDown('service', serviceOption);
      serviceOption - 1
      continueButtonClick();
    fi
   error:
      log("no options left, test failed")
  done
}
```
Where we iterate through all the dropdown options seeing if any will work, logging and erroring if none do

Further reading and other suggestions on dealing with conditional testing can be found here: https://docs.cypress.io/guides/core-concepts/conditional-testing